### ISSUE TITLE
[BUG] Permission set incorrect template folder .sh files 

### ISSUE THREAD

**Issue by alber-khan** [hestiacp/hestiacp#1845]
**Describe the bug**
nginx is crashing after updating the Hestia version to 1.4.1

**To Reproduce**
What steps did you take when the issue occurred? 
1. Click on the "Web" tab.
2. Click on "Edit Web Domain".
![image](https://user-images.githubusercontent.com/39749258/119807885-329db180-bf01-11eb-9f10-0255f0f3a236.png)
3. change the Proxy Template to caching.

**Expected behaviour**
should be able to change the Nginx template without any issue.

**Screenshots**
![image](https://user-images.githubusercontent.com/39749258/119807599-eb172580-bf00-11eb-92be-0923152a77fa.png)

**Operating system:**
Ubuntu 20.04 LTS

**Hestia Control Panel version:**
v1.4.1

**Additional context**
nginx: [emerg] "proxy_cache" zone "domain.com" is unknown in /etc/nginx/nginx.conf:176
nginx: configuration file /etc/nginx/nginx.conf test failed
Job for nginx.service failed because the control process exited with error code.
See "systemctl status nginx.service" and "journalctl -xe" for details.

this is what I got on my email

**Comment by jaapmarcus** [comment]
Check in /etc/nginx/conf.d/01_caching_pool.conf 

proxy_cache_path /var/cache/nginx/domain.com levels=2 keys_zone=domain.com:10m inactive=60m max_size=512m;

**Comment by alber-khan** [comment]
there is no `01_caching_pool.conf` file in `/etc/nginx/conf.d` directory. 

![image](https://user-images.githubusercontent.com/39749258/119817521-b5c40500-bf0b-11eb-8ca9-8d546f216c1b.png)

**Comment by jaapmarcus** [comment]
https://github.com/hestiacp/hestiacp/blob/main/install/deb/templates/web/nginx/caching.sh

What are the permissions of /usr/local/hesita/data/templates/web/nginx/caching.sh

**Comment by alber-khan** [comment]
`/usr/local/hestia/data/templates/web/nginx/caching.sh`  has the permission of '644'

**Comment by jaapmarcus** [comment]
Should be 755  chmod them and then try again

**Comment by alber-khan** [comment]
tired both 755 and 777 permission both aren't working and getting this email 

![image](https://user-images.githubusercontent.com/39749258/119825531-ab5a3900-bf14-11eb-899b-b92076a695e6.png)

**Comment by jaapmarcus** [comment]
v-update-web-domain-proxy-tpl user domain cacheing

**Comment by alber-khan** [comment]
i think it's best if we add permission validation on the upgrade script.

**Comment by alber-khan** [comment]
this issue happens with 3 servers after an automatic update. all the server went down and I gotta change the Proxy Template to `default` in more than 30 domains!

**Comment by ayoubamine** [comment]
I can't reach any external URL!
```bash
ping google.com
ping: google.com: Temporary failure in name resolution
```

**Comment by jaapmarcus** [comment]
https://forum.hestiacp.com/t/hestiacp-very-slow-dns-errors-ssl-errors-hetzner/2585/20
or https://github.com/hestiacp/hestiacp/issues/1849

**Comment by alber-khan** [comment]
I've also noticed `Purge Nginx Cache' button isn't working.

**Comment by jaapmarcus** [comment]
issue is found here
https://github.com/hestiacp/hestiacp/blob/bd64dab81c525787fa75022259d74b2d9ca5e9fe/src/hst_autocompile.sh#L621-L628

**Comment by jaapmarcus** [comment]
Fixed for 1.4.2
